pkg/controller/ipam/validate.go

```diff
@@ -50,14 +51,47 @@ func ValidateIPPool(review *admv1.AdmissionReview) *admv1.AdmissionResponse {
 	case admv1.Create:
 		// This shouldn't happen with the webhook configuration we include in the Antrea YAML manifests.
 		klog.V(2).Info("Validating CREATE request for IPPool")
-		// Always allow CREATE request.
+
+		// Validate individual ranges
+		for _, r := range newObj.Spec.IPRanges {
+			allowed, msg = validateIPRange(r)
+			if !allowed {
+				return validationResult(allowed, msg)
+			}
+		}
+
+		// Validate that the added ipRanges do not overlap
+		for i, r1 := range newObj.Spec.IPRanges[0 : len(newObj.Spec.IPRanges)-1] {
```

Comment:
I think this can be omitted, as it covers the whole range?

Reply:
Can you clarify? I'm not sure I understand. The 1st loop validates each range individually. The 2nd loop checks for overlaps - therefore loops from the 1st ipRange to the n-1 range. The inner loop runs from i+1 till the n-th as no need to recheck whatever was already checked before.

Reply:
Sorry, I didn't phrase the comment well. I meant to suggest that `newObj.Spec.IPRanges[0 : len(newObj.Spec.IPRanges)-1]` is equivalent to `newObj.Spec.IPRanges`

Reply:
I think that it covers the whole range, exclude the last item. As the inner loop will reach it and compare the Nth item with the N-1th

Reply:
Sorry, my mistake - you are right!